File: python/Zetcode/Advanced PyQt4 (PySide) Tutorial/05 - Layout Management/buttons.py
#!/usr/bin/env python

import sys
from PySide import QtGui, QtCore

class Example(QtGui.QWidget):

    def __init__(self):
        QtGui.QWidget.__init__(self)

        self.setGeometry(300, 300, 300, 150)
        self.setWindowTitle("Buttons")
        self.initUI()

    def initUI(self):
        ok = QtGui.QPushButton("OK")
        cancel = QtGui.QPushButton("Cancel")

        hbox = QtGui.QHBoxLayout()
        hbox.addStretch(1)
        hbox.addWidget(ok)
        hbox.addWidget(cancel)

        vbox = QtGui.QVBoxLayout()
        vbox.addStretch(1)
        vbox.addLayout(hbox)

        self.setLayout(vbox)


if __name__ == "__main__":
    app = QtGui.QApplication([])
    root = Example()
    root.show()
    sys.exit(app.exec_())

File: cpp/ZetCode/Qt4/07-LayoutManagement/buttons.cpp
#include "buttons.h"
#include <QtGui/QVBoxLayout>
#include <QtGui/QHBoxLayout>


Buttons::Buttons(QWidget *parent)
    : QWidget(parent) {
    QVBoxLayout *mainLayout = new QVBoxLayout(this);
    QHBoxLayout *buttonLayout = new QHBoxLayout();

    okButton = new QPushButton("OK", this);
    applyButton = new QPushButton("Apply", this);

    buttonLayout->addWidget(okButton, 1, Qt::AlignRight);
    buttonLayout->addWidget(applyButton, 0, Qt::AlignRight);

    mainLayout->addStretch(1);
    mainLayout->addLayout(buttonLayout);
}
